design/0001-frr.md

```diff
@@ -0,0 +1,347 @@
+│               │                       │               │
+└───────────────┘                       └───────▲───────┘
+                                                │<<implements>>
+                                     ┌──────────┴─────────┐
+                                     │                    │
+                              ┌──────┴────────┐  ┌────────┴──────┐
+                              │     main::    │  │    main::     │
+                              │ L2_controller │  │ bgp_controller│
+                              │   <<class>>   │  │   <<class>>   │
+                              │               │  │               │
+                              └───────────────┘  └───────┬───────┘
+                                                         │
+                                                         │  <<use>>
+                                                 ┌───────▼───────┐
+                                                 │               │
+                                                 │    session    │
+                                                 │ <<interface>> │
+                                                 │               │
+                                                 └───────▲───────┘
+                                                         │ <<implement>>
+                                             ┌───────────┴─────────────┐
+                                             │                         │
+                                     ┌───────┴───────┐         ┌───────┴───────┐
+                                     │               │         │               │
+                                     │  bgp_metallb  │         │    bgp_frr    │
+                                     │  <<package>>  │         │  <<package>>  │
+                                     │               │         │               │
+                                     └───────────────┘         └───────┬───────┘
+                                                                       │
+                                                                       │
+                                                               ┌───────┴───────┐
+                                                               │               │
+                                                               │      frr      │
+                                                               │ <<container>> │
+                                                               │               │
+                                                               └───────────────┘
+
+```
+
+#### FRR deployment
+
+FRR will be deployed as a container. The speaker application will be configured
+at startup to use the FRR bgp daemon and will provide details for connection to
+the daemon (e.g. location of a socket)
+
+The container will expose a control interface that will initially be used to
+configure FRR via the `vtysh` application and eventually via its gRPC golang
```

Comment:
are u planning to use vtysh to configure FRR or the standard MetalLB configmap and in future CRD ?
if we will use k8s way to configure do we need to disable vtysh to change the config ?

Reply:
I have no plans to allow direct configuration of FRR via the MetalLB ConfigMap (or CRD).  There may be use-cases in which configuration of the FRR instance will need to be shared between two actors (MetalLB and another) but in that case we will partition FRR appropriately.

Reply:
In fact, I'd call out keeping FRR as an implementation detail as an explicit design goal.  What do you think, Mark?